runtime/stdlib/test.go

```diff
@@ -1367,6 +1428,280 @@ var equalMatcherFunction = interpreter.NewUnmeteredHostFunctionValue(
+		Parameters: []sema.Parameter{
+			{
+				Label:      sema.ArgumentLabelNotRequired,
+				Identifier: "count",
+				TypeAnnotation: sema.NewTypeAnnotation(
+					sema.IntType,
+				),
+			},
+		},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(matcherType),
+	}
+}()
+
+var haveElementCountMatcherFunction = interpreter.NewUnmeteredHostFunctionValue(
+	haveElementCountMatcherFunctionType,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		count, ok := invocation.Arguments[0].(interpreter.IntValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		haveElementCountTestFunc := interpreter.NewHostFunctionValue(
+			nil,
+			matcherTestFunctionType,
+			func(invocation interpreter.Invocation) interpreter.Value {
+				var matchingCount bool
+				switch value := invocation.Arguments[0].(type) {
+				case *interpreter.ArrayValue:
+					matchingCount = value.Count() == count.ToInt(invocation.LocationRange)
+				case *interpreter.DictionaryValue:
+					matchingCount = value.Count() == count.ToInt(invocation.LocationRange)
+				default:
+					panic(errors.NewDefaultUserError("expected Array or Dictionary argument"))
+				}
+
+				return interpreter.AsBoolValue(matchingCount)
+			},
+		)
+
+		return newMatcherWithGenericTestFunction(invocation, haveElementCountTestFunc)
+	},
+)
+
+const containMatcherFunctionName = "contain"
+
+const containMatcherFunctionDocString = `
+Returns a matcher that succeeds if the tested value is an array that contains
+a value that is equal to the given value, or the tested value is a dictionary
+that contains an entry where the key is equal to the given value.
+`
+
+var containMatcherFunctionType = func() *sema.FunctionType {
+	return &sema.FunctionType{
+		IsConstructor:  false,
+		TypeParameters: []*sema.TypeParameter{},
+		Parameters: []sema.Parameter{
+			{
+				Label:      sema.ArgumentLabelNotRequired,
+				Identifier: "element",
+				TypeAnnotation: sema.NewTypeAnnotation(
+					sema.AnyStructType,
+				),
+			},
+		},
+		ReturnTypeAnnotation: sema.NewTypeAnnotation(matcherType),
+	}
+}()
+
+var containMatcherFunction = interpreter.NewUnmeteredHostFunctionValue(
+	containMatcherFunctionType,
+	func(invocation interpreter.Invocation) interpreter.Value {
+		element, ok := invocation.Arguments[0].(interpreter.EquatableValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+
+		inter := invocation.Interpreter
+
+		containTestFunc := interpreter.NewHostFunctionValue(
+			nil,
+			matcherTestFunctionType,
+			func(invocation interpreter.Invocation) interpreter.Value {
+				var elementFound interpreter.BoolValue
+				switch value := invocation.Arguments[0].(type) {
+				case *interpreter.ArrayValue:
+					elementFound = value.Contains(
+						inter,
+						invocation.LocationRange,
+						element,
+					)
+				case *interpreter.DictionaryValue:
+					elementFound = value.ContainsKey(
+						inter,
+						invocation.LocationRange,
+						element,
+					)
```

Comment:
Maybe this should check both keys and values? Then again, e.g. [testify](https://github.com/stretchr/testify)'s `assert.Contains` also only checks map keys 🤔